File: backend/ShopBackend/Models/ProductImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShopBackend.Models
{
    [Table("product_images")]
    public class ProductImage
    {
        [Key]
        public long Id { get; set; }

        [Column("product_id")]
        public long ProductId { get; set; }

        [Column("image_url")]
        public string ImageUrl { get; set; }

        [Column("is_main")]
        public bool IsMain { get; set; }

    }
}

File: app/shopapp/lib/models/product_image.dart
class ProductImage {
  final int id;
  final int productId;
  final String imageUrl;
  final bool isMain;

  ProductImage({
    required this.id,
    required this.productId,
    required this.imageUrl,
    required this.isMain,
  });

  factory ProductImage.fromJson(Map<String, dynamic> json) {
    return ProductImage(
      id: json["id"],
      productId: json["product_id"],
      imageUrl: json["image_url"] ?? "",
      isMain: json["is_main"] == 1 || json["is_main"] == true,
    );
  }
}
